### PR TITLE
Armoryd: TX handling updates.

### PR DESCRIPTION
**Pull request by STRML**
This extends #35 without breaking backcompat.

Armoryd: TX handling updates.

Changes:

- Accept raw TX strings in addition to paths.
  - Added *raw() versions of sendasciitransaction(),
    signasciitransaction(), and gethextxtobroadcast().
    - These functions take actual ascii input, not a file name.
  - Added readFile() helper that handles read error.

- Add decodeasciitransaction() and decodeasciitransactionraw(),
  which returns json.

- Added 'ASCII' output type to getlockboxinfo().

### CODE REVIEW

File: armoryd.py [hunk 1]
```diff
@@ -2012,19 +2024,97 @@ def create_unsigned_transaction(self, scriptValuePairs, spendFromLboxID=None, wa
...
+      if ustxObj:
+         return ustxObj.toJSONMap()
+
+      return
+
+   #############################################################################
+   # Take the path of the ASCII representation of an unsigned Tx and sign it.
+   # See `signasciitransactionraw` for more details.
+   @catchErrsForJSON
+   def jsonrpc_signasciitransaction(self, txASCII):
```
Comment (goatpig) [inline armoryd.py]: this should be txASCIIFile

**Comment by goatpig** [comment]
have you tested this?

File: armoryd.py (hunk 1 above)
Comment (STRML) [inline armoryd.py]: Thanks fixed

**Comment by STRML** [comment]
> have you tested this?

Not before I submitted the PR, but yes, we're using this for some signing now & noticed the param name bug when we did. No problems so far.

**Comment by goatpig** [comment]
Will merge it in sometimes this week. Please comment on PR #80:

https://github.com/goatpig/BitcoinArmory/pull/80

**Pull request merged**